earth2studio/models/dx/cyclone_tracking.py

```diff
@@ -0,0 +1,1063 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+from typing import Any
+
+import cupy as cp
+import numpy as np
+import torch
+from cucim.skimage.feature import peak_local_max as cucim_peak_local_max
+from cucim.skimage.measure import label, regionprops
+from cucim.skimage.morphology import binary_erosion, remove_small_objects
+from pandas import DataFrame
+from skimage.feature import peak_local_max as skimage_peak_local_max
+from skimage.morphology import convex_hull_image
+
+from earth2studio.models.auto import AutoModelMixin, Package
+from earth2studio.models.batch import batch_coords, batch_func
+from earth2studio.models.dx.base import DiagnosticModel
+from earth2studio.utils import (
+    handshake_coords,
+    handshake_dim,
+)
+from earth2studio.utils.type import CoordSystem
+
+VARIABLES_TC = [
+    "u850",
+    "v850",
+    "msl",
+    "z500",
+    "z850",
+    "z200",
+    "t500",
+    "t400",
+    "t300",
+    "t250",
+    "t200",
+]
+
+VARIABLES_TCV = ["u850", "v850", "u10m", "v10m", "msl"]
+
+OUT_VARIABLES = ["tc_lat", "tc_lon", "tc_msl", "tc_w10m"]
+
+
+class CycloneTrackingVorticity(torch.nn.Module, AutoModelMixin):
+    """
+
+    Finds a list of tropical cyclone centers using an adaption of the method described in the conditions
+    in Wu and Duan 2023 vort850
+
+    https://doi.org/10.1016/j.wace.2023.100626.
```

Comment:
Nit: Add these links in a Note block so they get called out better in the dos its excellent that these are included!

E.g. https://github.com/NVIDIA/earth2studio/blob/main/earth2studio/models/px/dlwp.py#L50